lib/em/pure_ruby.rb

```diff
@@ -180,21 +180,22 @@ def set_tls_parms signature, priv_key, cert_chain, verify_peer, fail_if_no_peer_
       unless dhparam.nil? || dhparam.empty?
         raise "set_tls_params dhparam has not been implemented in pure Ruby"
       end
-      ssl_version, _ = {
-        :TLSv1_2 => EM_PROTO_TLSv1_2,
-        :TLSv1_1 => EM_PROTO_TLSv1_1,
-        :TLSv1 => EM_PROTO_TLSv1,
-        :SSLv3 => EM_PROTO_SSLv3,
-        :SSLv23 => EM_PROTO_SSLv2
-      }.detect do |_, bits|
-        (bits & protocols_bitmask) != 0
+      ssl_options = {
+        EM_PROTO_SSLv2 => OpenSSL::SSL::OP_NO_SSLv2,
+        EM_PROTO_SSLv3 => OpenSSL::SSL::OP_NO_SSLv3,
+        EM_PROTO_TLSv1 => OpenSSL::SSL::OP_NO_TLSv1,
+        EM_PROTO_TLSv1_1 => OpenSSL::SSL::OP_NO_TLSv1_1,
+        EM_PROTO_TLSv1_2 => OpenSSL::SSL::OP_NO_TLSv1_2
+      }.inject(0) do |options, (bits, option)|
+        options |= option if (bits & protocols_bitmask) == 0
+        options
```

Comment:
Yes, this better matches the C++ by looping over the entire array. Note that `ssl_version` might also be a non-array (but I think I coerce it to an array in lib/em/connection.rb, so by the time it gets to the reactor it should always be an array? gotta double check).

This is something I implemented differently in EM 1.2 vs. how Ruby does it, is allowing an array of possible `ssl_version`s. I want to propose this upstream; the way Ruby does it now makes it fairly hard to write "Allow the following SSL/TLS protocol versions: TLS 1.0, 1.1, or 1.2"